packages/backend-function/src/factory.ts

```diff
@@ -209,14 +224,49 @@ class AmplifyFunction
   readonly resources: FunctionResources;
   constructor(scope: Construct, id: string, props: HydratedFunctionProps) {
     super(scope, id);
+    let bannerCode;
+    const hasSecrets = secretPaths.length > 0;
+
+    if (hasSecrets) {
```

Comment:
any reason that we can't just add this unconditionally to the lambda bundle? If there are no secrets, then it's a noop

Reply:
I'm thinking for users who do not have any secrets they may be wondering why there is a code snippet in their lambda, especially since it's hard to read being condensed into one line.

Reply:
True, but do we expect anyone to go poking in the bundled lambda code? We can also include a comment in the banner explaining what the code does and that it's a noop if there are no secrets.

Reply:
That's true, I believe that is unlikely unless prompted to by us. Adding a comment at the top.